src/util/async/AnyExecutionContext.hpp

```diff
@@ -221,10 +246,49 @@ class AnyExecutionContext {
     };
 
     template <typename CtxType>
-    struct Model : Concept {
+    struct OwningModel : Concept {
```

Comment:
Why do you need `OwningModel` and `NonOwningModel`? `CtxType` is a template parameter anyway, so you could pass it here as a reference or move.